metagraph/src/graph/annotated_dbg.cpp

```diff
@@ -49,6 +50,11 @@ bool AnnotatedDBG::check_compatibility() const {
     // TODO: find a way to fix this thing to add check_compatibility() to the constructor
     // of AnnotatedSequenceGraph.
     // Maybe make it how it was before 9f4e8e5b73571d874dd93eedc53079f71615023c
+
+    // TODO: what if CanonicalDBG is not the highest level? find a better way to do this
+    if (const auto *canonical = dynamic_cast<const CanonicalDBG*>(&dbg_))
+        return canonical->get_graph().max_index() == annotator_->num_objects();
```

Comment:
CanonicalDBG should not be passed to this class. It should always be the base graph.

Reply:
Ah, never mind. Looks like we've been passing canonical_dbg to this since it was added.